public/controllers/register-agent/hooks/use-register-agent-commands.ts

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import { CommandGenerator } from '../core/register-commands/command-generator/command-generator';
+import {
+  IOSDefinition,
+  IOperationSystem,
+  IOptionalParameters,
+  tOptionalParams,
+} from '../core/register-commands/types';
+
+interface IUseRegisterCommandsProps<OS extends IOperationSystem, Params extends string> {
+  osDefinitions: IOSDefinition<OS, Params>[];
+  optionalParamsDefinitions: tOptionalParams<Params>;
+}
+
+interface IUseRegisterCommandsOutput<OS extends IOperationSystem, Params extends string> {
+  selectOS: (params: OS) => void;
+  setOptionalParams: (params: IOptionalParameters<Params>) => void;
+  installCommand: string;
+  startCommand: string;
+  optionalParamsParsed: IOptionalParameters<Params> | {};
+}
+
+
+/**
+ * Custom hook that generates install and start commands based on the selected OS and optional parameters.
+ * 
+ * @template T - The type of the selected OS.
+ * @param {IUseRegisterCommandsProps} props - The properties to configure the command generator.
+ * @returns {IUseRegisterCommandsOutput<OS,Params>} - An object containing the generated commands and methods to update the selected OS and optional parameters.
+ */
+export function useRegisterAgentCommands<OS extends IOperationSystem, Params extends string>(props: IUseRegisterCommandsProps<OS, Params>): IUseRegisterCommandsOutput<OS,Params> {
+  const { osDefinitions, optionalParamsDefinitions } = props;
+  // command generator settings
+  const wazuhVersion = '4.4';
```

Comment:
is it ok to be hardcoded?

Reply:
It should not be hardcoded.

Reply:
Yes, that's right. I will change it to get the current wazuh version